docs: update PR template (#896)
hyperledger-labs/open-enterprise-agent: .github/PULL_REQUEST_TEMPLATE.md

### Description: 
Summarize the changes you're submitting in a few sentences, including Jira ticket ATL-xxxx if applicable.

### Context: 

> It creates traceability on the matter tackled by the PR. It will also help the reviewer understand what the PR is about ---- linking to related issues, bug reports, or discussions for context.

Reviewer: ```suggestion
It creates traceability on the matter tackled by the PR. It will also help the reviewer understand the PR, linking to related issues, bug reports, or discussions for context.
```

### Alternatives Considered (optional): 

> Link to existing ADR if any. If relevant, describe other approaches explored and why this approach was chosen. Documenting why certain paths were not taken creates a knowledge base for future reference, helping prevent others from revisiting less optimal ideas.

Reviewer: ```suggestion
Link to existing ADR, if any. If relevant, describe other approaches explored and the selected approach. Documenting why the methods were not selected will create a knowledge base for future reference, helping prevent others from revisiting less optimal ideas.
```

Reviewer: Also define what ADR is

Author: @petevielhaber , I have explicated the acronym ADR. Please check if good enough.

### Checklist: 
- [ ] My PR follows the contribution guidelines of this project

> - [ ] My PR is free of third-party dependencies that don't comply with HL-Allowlist

Reviewer 2: - [ ] My PR is free of third-party dependencies that don't comply with the [Allowlist](https://toc.hyperledger.org/governing-documents/allowed-third-party-license-policy.html#approved-licenses-for-allowlist)
